### PR TITLE
PAT-362 : View and manage active interventions

### PR DESCRIPTION
**Pull request by timharrison-moj**
* This extends the actions screen with the display of controls and interventions.
* It provides for status changes & completed/current items in separate tabs
* Includes unit tests for the main functionality of allocating, removing and merging active C&I
* Includes refactoring of C&I for better module separation and tidiness.

### CODE REVIEW

File: server/routes/controlsAndInterventions.js [hunk 1]
```diff
@@ -135,15 +131,17 @@ module.exports = function ctrlAndIntRoutes({ offenderService, pathfinderService
...
       if (!isNilOrEmpty(overrides) && overrides.length > 0) {
         res.render('pages/cim/overrides', {
           offenderBasic,
           data: { nominalId, riskBandId, needsId, controlType, overrides },
         })
       } else {
-        const mergeOk = await pathfinderService.mergeCimControlsToActive(requestForm, user)
-        logger.info(`Controls POST merge ${mergeOk}`)
+        const mergeOk = await pathfinderService.mergeCimControlsToActive(identifiers)
+        logger.debug(`Controls POST merge ${mergeOk}`)
```
Comment (garethmdavies) [inline server/routes/controlsAndInterventions.js]: Seems to be a mix of logging at info and debug level in this file.

Comment (timharrison-moj) [reply server/routes/controlsAndInterventions.js]: I can sort that as a follow-on piece of work. I am also looking at standardising the way in which any DB errors are thrown up to be handled by the application. Quite a few places don't catch/throw errors as yet.

File: server/data/pathfinderClient.js [hunk 2]
```diff
@@ -703,6 +703,30 @@ const getActiveControls = async (nominalId, type) => {
   return result.rows
 }
 
+const getActiveControlsByStatus = async (nominalId, statusOne, statusTwo, statusThree) => {
```
Comment (lbennett-moj) [inline server/data/pathfinderClient.js]: maybe use array param and ANY 
eg status = ANY ($2)

File: server/views/pages/actions/summary.html [hunk 3]
```diff
@@ -50,66 +78,252 @@ <h1 class="govuk-heading-l">Actions</h1>
...
+        <input type="hidden" name="_csrf" value="{{ csrfToken }}" />
+        <input type="hidden" name="nominalId" value="{{ data.nominalId }}" />
+
+      {% set rowCount = 0 %}
+
+      {% for control in data.currentControls %}
+
+        {# -- Set the alternating row shading style #}
+        {% set rowCount = rowCount + 1 %}
+
```
Comment (steverendell) [inline server/views/pages/actions/summary.html]: You can use cycler: https://mozilla.github.io/nunjucks/templating.html#cycler-item1-item2-itemn for this :)

File: server/views/pages/actions/summary.html [hunk 4]
```diff
@@ -50,66 +78,252 @@ <h1 class="govuk-heading-l">Actions</h1>
...
+        {% if rowCount % 2 === 0 %}
+          <tr class="govuk-table__row govuk-!-font-size-19" style="background-color: #f3f2f1;">
+        {% else %}
+          <tr class="govuk-table__row  govuk-!-font-size-19">
+        {% endif %}
+
+        <td class="govuk-table__cell">{{ control.description }}</td>
+        <td class="govuk-table__cell">{{ mapControlType(control.controlType) }}</td>
+        <td class="govuk-table__cell">
+        {% if control.mandatory === 'Yes' %}
```
Comment (steverendell) [inline server/views/pages/actions/summary.html]: should this be moved to a css class?

File: server/data/pathfinderClient.js (hunk 2 above)
Comment (timharrison-moj) [reply server/data/pathfinderClient.js]: Thanks Lucy - that's exactly what I was looking for - and did't find! The first version passed a list o status values, but couldn't get the SQL to correctly recognise for a IN ($2) statement. I'll do this change now.

File: server/routes/controlsAndInterventions.js (hunk 1 above)
Comment (timharrison-moj) [reply server/routes/controlsAndInterventions.js]: ok, addressed this one.

File: server/views/pages/actions/summary.html (hunk 4 above)
Comment (timharrison-moj) [reply server/views/pages/actions/summary.html]: Yes, probably. In fact, I've just discovered another issue with this - when column sorting by different columns the row-shading gets mixed up, so looking at different ways now.

File: server/data/pathfinderClient.js (hunk 2 above)
Comment (timharrison-moj) [reply server/data/pathfinderClient.js]: Thanks. Done this one now. I was trying the where statusCode in($2) with a list, and it did NOT like that, but its happy with = ANY(<list>)

File: server/views/pages/actions/summary.html (hunk 3 above)
Comment (timharrison-moj) [reply server/views/pages/actions/summary.html]: That's a cracking tip- thanks Steve. Didn't know that existed!

File: server/views/pages/actions/summary.html (hunk 4 above)
Comment (timharrison-moj) [reply server/views/pages/actions/summary.html]: I've changed to use the nunjucks cycler and CSS class for the row style. Also looking at the column sorter JS file to see if I can juggle the rowclasses dynamically.

File: server/data/pathfinderClient.js (hunk 2 above)
Comment (lbennett-moj) [reply server/data/pathfinderClient.js]: I'll slack you an example from cat tool just for info

**Pull request merged**